File: examples/competitors.ts
// Company competitor analysis
import { VeroqClient } from '@veroq/sdk';

const client = new VeroqClient();
const ticker = process.argv[2] || 'TSLA';
const result = await client.ask(`Who are ${ticker}'s main competitors and how do they compare?`);

console.log(result.summary);

if (result.trade_signal) {
  const { action, score } = result.trade_signal;
  console.log(`\n${ticker} signal: ${action} (${score}/100)`);
}

File: examples/competitors.py
"""Find a company's main competitors."""
from veroq import VeroqClient

client = VeroqClient()
result = client.ask("Who are NVDA's main competitors?")

data = result.get("data", {})
competitors = data.get("competitors", {}).get("companies", [])

if competitors:
    print("NVDA Competitors\n")
    print(f"  {'Company':<20} {'Ticker':>6} {'Price':>10} {'Mkt Cap':>14}")
    print(f"  {'-'*54}")
    for c in competitors:
        name = c.get("name", "?")
        ticker = c.get("ticker", "?")
        price = c.get("price", 0)
        mcap = c.get("market_cap", 0)
        print(f"  {name:<20} {ticker:>6} ${price:>9,.2f} ${mcap:>13,.0f}")
else:
    print(result.get("summary", "No competitor data available."))
